packages/flutter_tools/gradle/src/main/kotlin/FlutterPluginUtils.kt

```diff
@@ -371,4 +392,433 @@ object FlutterPluginUtils {
         // doesn't support.
         return project.property(PROP_LOCAL_ENGINE_BUILD_MODE) == flutterBuildMode
     }
+
+    private fun getAndroidExtension(project: Project): BaseExtension {
+        // Common supertype of the android extension types.
+        // But maybe this should be https://developer.android.com/reference/tools/gradle-api/8.7/com/android/build/api/dsl/TestedExtension.
+        return project.extensions.findByType(BaseExtension::class.java)!!
+    }
+
+    @JvmStatic
+    @JvmName("getCompileSdkFromProject")
+    internal fun getCompileSdkFromProject(project: Project): String = getAndroidExtension(project).compileSdkVersion!!.substring(8)
```

Comment:
Can you add a comment covering where substring(8) comes from and maybe what the expected data would look like.